src/actions/textobject.ts

```diff
@@ -512,6 +512,53 @@ export class SelectInnerParagraph extends TextObjectMovement {
   }
 }
 
+@RegisterAction
+export class SelectEntire extends TextObjectMovement {
+  keys = ['a', 'e'];
+
+  public async execAction(position: Position, vimState: VimState): Promise<IMovement> {
+    let lines: number = TextEditor.getLineCount();
+    let lastLine = lines > 0 ? lines - 1 : 0;
+
+    return {
+      start: new Position(0, 0),
+      stop: new Position(lastLine, TextEditor.getLineLength(lastLine)),
+    };
+  }
+}
+
+@RegisterAction
+export class SelectEntireIgnoringLeadingTrailing extends TextObjectMovement {
+  keys = ['i', 'e'];
+
+  public async execAction(position: Position, vimState: VimState): Promise<IMovement> {
+    let lines: number = TextEditor.getLineCount();
+    let start: Position = new Position(0, 0);
+    let stop: Position = new Position(lines - 1, 0).getLineEnd();
```

Comment:
Please use `Position.getDocumentEnd()`.

If you'd like to do a small refactor (not necessary), you could move this and `Position.getDocumentBegin()` to `TextEditor`, since there's no reason for them to be in `Position`.

Reply:
✔️